libraries/common/set.effekt

```diff
@@ -0,0 +1,208 @@
+module set
+
+import map
+
+// Ordered finite set, backed by a `Map`.
+record Set[A](internal: Map[A, Unit])
+
+// Create a new empty set.
+//
+// O(1)
+def empty[A](): Set[A] = {
+  Set(empty())
+}
+
+// Check if set `s` is empty.
+//
+// O(1)
+def isEmpty[A](s: Set[A]): Bool = {
```

Comment:
maybe we could use `empty?` here?

Reply:
Thanks!
In general, feel free to suggest better names everywhere, I really just took the first idea I had every time :)

Reply:
We should do this change consistently across the stdlib, if we decide for it